src/v/kafka/server/quota_manager.h

```diff
@@ -58,7 +65,60 @@ class quota_manager : public ss::peering_sharded_service<quota_manager> {
         clock::duration duration{0};
     };
 
-    quota_manager();
+    using bucket_t = atomic_token_bucket<ss::internal::capped_release::yes>;
+
+    /// A thread-safe reference wrapper to help safely share a thread-safe value
+    /// across cores. It is safe to hold on to this reference across scheduling
+    /// points. This wrapped ensures that deallocation happens on the core that
+    /// created this wrapper.
+    template<typename T>
+    class value_node {
+    public:
+        value_node(std::unique_ptr<T> unq)
+          : _value{std::make_shared<ss::foreign_ptr<std::unique_ptr<T>>>(
+            std::move(unq))} {}
+
+        /// Takes a non-owning reference to the wrapped value
+        /// The lifetime of the reference is at least the lifetime of the node.
+        T& operator*() const { return &*_value.get()->get(); }
+
+        /// Takes a non-owning pointer to the wrapped value
+        /// The lifetime of the pointer is at least the lifetime of the node.
+        T* operator->() const { return _value.get()->get(); }
+
+    private:
+        // TODO: replace std::shared_ptr with ssx::sharded_ptr to reduce
+        // cross-core contention on the reference count
+        // Note: the unique_ptr is only needed because ss::foreign_ptr expects
+        // to wrap a pointer type
+        std::shared_ptr<ss::foreign_ptr<std::unique_ptr<T>>> _value;
+    };
+
+    template<typename T, typename... Args>
+    value_node<T> make_value_node(Args&&... args) {
+        return value_node<T>(std::make_unique<T>(std::forward<Args>(args)...));
+    }
+
+    // Accounting for quota on per-client and per-client-group basis
+    // last_seen: used for gc keepalive
+    // tp_rate: throughput tracking
+    // pm_rate: partition mutation quota tracking - only on home shard
+    struct client_quota {
+        ssx::sharded_value<clock::time_point> last_seen_ms;
+        // TODO: do we need to pad these?
+        std::optional<bucket_t> tp_produce_rate;
+        std::optional<bucket_t> tp_fetch_rate;
+        std::optional<token_bucket_rate_tracker> pm_rate;
```

Comment:
I think it's worth doing this one, too, and removing the dispatch to home core.  And then `token_bucket_rate_tracker` can be removed.

Reply:
I've made this change now